pajbot/apiwrappers/base.py

```diff
@@ -73,25 +73,34 @@ def join_base_and_endpoint(base, endpoint):
         else:
             return BaseAPI.join_base_and_string(base, endpoint)
 
-    def request(self, method, endpoint, params, headers, json=None, **request_options):
-        full_url = self.join_base_and_endpoint(self.base_url, endpoint)
+    def request(self, method, endpoint, params, headers, json=None, base_url=None, data=None, **request_options):
+        if base_url is None:
+            base_url = self.base_url
+        full_url = self.join_base_and_endpoint(base_url, endpoint)
         response = self.session.request(
-            method, full_url, params=params, headers=headers, json=json, timeout=self.timeout, **request_options
+            method,
+            full_url,
+            params=params,
+            headers=headers,
+            json=json,
+            timeout=self.timeout,
+            data=data,
```

Comment:
The data parameter should also not be needed, since we have the request_options "catchall" token, especially if it's only for a single use case